core/src/main/java/feast/core/model/FeatureSource.java

```diff
@@ -0,0 +1,202 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright 2018-2020 The Feast Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package feast.core.model;
+
+import static feast.proto.core.FeatureSourceProto.FeatureSourceSpec.SourceType.*;
+
+import feast.core.util.TypeConversion;
+import feast.proto.core.FeatureSourceProto.FeatureSourceSpec;
+import feast.proto.core.FeatureSourceProto.FeatureSourceSpec.BigQueryOptions;
+import feast.proto.core.FeatureSourceProto.FeatureSourceSpec.FileOptions;
+import feast.proto.core.FeatureSourceProto.FeatureSourceSpec.FileOptions.FileFormat;
+import feast.proto.core.FeatureSourceProto.FeatureSourceSpec.KafkaOptions;
+import feast.proto.core.FeatureSourceProto.FeatureSourceSpec.KinesisOptions;
+import feast.proto.core.FeatureSourceProto.FeatureSourceSpec.SourceType;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.EnumType;
+import javax.persistence.Enumerated;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import javax.persistence.Table;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.Setter;
+
+@Entity
+@Getter
+@Setter(AccessLevel.PRIVATE)
+@Table(name = "feature_sources")
+public class FeatureSource {
```

Comment:
This table is going to become crazy big if we keep it denormalized, but I guess its ok for now.

Reply:
Can we not flatten all the fields?
I'm for the option with kafkaConfig, bqConfig, fileConfig columns instead with json inside

Reply:
So each of those is a map?

Reply:
Updated to only using `config` column.